packages/astro/src/runtime/client/dev-overlay/ui-library/icons.ts

```diff
@@ -0,0 +1,28 @@
+export type DefinedIcon = keyof typeof icons;
+export type Icon = DefinedIcon | (string & NonNullable<unknown>);
+
+export function isDefinedIcon(icon: Icon): icon is DefinedIcon {
+	return icon in icons;
+}
+
+export function getIconElement(
+	name: keyof typeof icons | (string & NonNullable<unknown>)
+): SVGElement | undefined {
+	const icon = icons[name as keyof typeof icons];
+
+	if (!icon) {
+		return undefined;
+	}
+
+	const svgFragment = new DocumentFragment();
+	svgFragment.append(document.createRange().createContextualFragment(icon));
+
+	return svgFragment.firstElementChild as SVGElement;
+}
+
+const icons = {
```

Comment:
Could you use these as the icons for the overlay plugins so there is a single source of truth for the icons?